tests/integrated/test-dataiterator/runtest

```diff
@@ -6,59 +6,34 @@
 
 from __future__ import print_function
 try:
-  from builtins import str
+    from builtins import str
 except:
-  pass
+    pass
 from boututils.run_wrapper import shell, shell_safe, launch_safe, getmpirun
 from boutdata.collect import collect
 from sys import stdout, exit
 
 MPIRUN=getmpirun()
 
 print("Making DataIterator test")
-s, out = launch_safe("make > make.log",pipe=True)
-if s:
-  print("Failed to run make")
-  exit(s);
+s, out = shell_safe("make > make.log",pipe=True)
 
 flags = [""]
-#, "nsys=2", "nsys=5 periodic", "nsys=7 n=10"]
 
-code = 0 # Return code
 for nproc in [1,2,4]:
-  for mproc in [1,2,4,8]:
-    cmd = "./test_dataiterator"
+    for mproc in [1,2,4,8]:
+        cmd = "./test_dataiterator"
 
-    print("   %d processors...." % (nproc))
-    r = 0
-    for f in flags:
-        stdout.write("\tflags '"+f+"' ... ")
+        print("   %d processors...." % (nproc))
 
-        shell("rm data/BOUT.dmp.* 2> err.log")
+        for f in flags:
+            stdout.write("\tflags '"+f+"' ... ")
 
-        # Run the case
-        s, out = launch_safe(cmd+" "+f, runcmd=MPIRUN, nproc=nproc, pipe=True)
-        f = open("run.log."+str(nproc)+"."+str(mproc), "w")
-        f.write(out)
-        f.close()
-        r = r + 1
-        if s:
-          #stdout.write(out)
-          code=1
-          print("FAILED")
-        else:
-          
-          # Find out if it worked
-          #allpassed = collect("allpassed", path="data", info=False)
-          #if allpassed:
-          print("PASSED")
-          #else:
-          #  print("FAILED")
-          #  code = 1
+            shell("rm data/BOUT.dmp.* 2> err.log")
 
-if code == 0:
-    print(" => All data iteration tests passed")
-else:
-    print(" => Some failed tests")
+            # Run the case
+            _, out = launch_safe(cmd+" "+f, runcmd=MPIRUN, nproc=nproc, pipe=True)
+            with open("run.log."+str(nproc)+"."+str(mproc), "w") as f:
```

Comment:
Probably not really an issue but you're redefining `f` here which is initially the current flag value that we're iterating over.